### PR TITLE
Update enc/common/tools to the latest

### PR DESCRIPTION
**Pull request by tmat**
Fixes sign build issue.

### CODE REVIEW

File: eng/common/tools.ps1 [hunk 1]
```diff
@@ -436,7 +442,7 @@ function MSBuild() {
   $cmdArgs = "$($buildTool.Command) /m /nologo /clp:Summary /v:$verbosity /nr:$nodeReuse"
 
   if ($warnAsError) { 
-    $cmdArgs += " /p:TreatWarningsAsErrors=true" 
+    $cmdArgs += " /warnaserror /p:TreatWarningsAsErrors=true"
```
Comment (jaredpar) [inline eng/common/tools.ps1]: @sharwell as I believe we removed this for the formatting analyzer work.

Comment (333fred) [inline eng/common/tools.ps1]: Unrelated?

Comment (tmat) [reply eng/common/tools.ps1]: Unrelated to the signing issue, but this PR just takes the Arcade's version of tools.ps1.

Comment (tmat) [reply eng/common/tools.ps1]: We can't be modifying tools.ps1 in Roslyn. If we need to disable warnings as errors then we need to set `$warnAsError = $false` in `build.ps1`

Comment (jaredpar) [reply eng/common/tools.ps1]: Forgot this was a protected directory. Actually I think this is fine. We don't need to differentiate between warning and error anymore. The packages around code style are completely disabled in the specific runs where we care about this.

Comment (sharwell) [reply eng/common/tools.ps1]: @tmat Build in master is failing until this change is reverted.
